File: java/gpt/p105.java
import java.io.*;

// p105.java - create secret.txt and write base64 string; set owner-only perms (best-effort)
public class p105 {
    public static void main(String[] args) {
        String fname = "secret.txt";
        String msg = "U2FsdGVkX1/R+WzJcxgvX/Iw==";
        try (FileWriter fw = new FileWriter(fname)) { fw.write(msg); }
        catch (IOException e) { e.printStackTrace(); System.exit(1); }
        File f = new File(fname);
        f.setReadable(true, true);
        f.setWritable(true, true);
        System.out.println(fname);
    }
}

File: javascript/gpt/p105.js
// p105.js - write secret.txt with base64 payload and set restrictive perms
const fs = require('fs');
const path = 'secret.txt';
fs.writeFileSync(path, 'U2FsdGVkX1/R+WzJcxgvX/Iw==', { encoding: 'utf8' });
try { fs.chmodSync(path, 0o600); } catch (e) { console.error('chmod failed', e); }
console.log(path);
